### PR TITLE
feat: refactoring cwc-icon to be consistent with newer components

### PR DESCRIPTION
**Pull request by mathisscott**
• replaced .sr-only classname with shared .clr-sr-only classname
• deprecated classname-driven API (.is-solid, .has-alert, .has-badge, .is-green, .is-success, .is-red, .is-error, .is-danger, .is-warning, .is-blue, .is-info, .is-highlight, .is-white, .is-inverse, .has-badge--success, .has-badge--danger, .has-badge--info, .has-badge--error
• added 'solid' boolean attribute
• added 'inverse' boolean attribute
• added 'alert' boolean attribute
• added 'status' attribute
• added 'badge' attribute
• updated icon demo to use new properties
• split stateful badge property into badge boolean and badgeType string properties
• updated unit tests
• created deprecated icons demo page to validate 3.0 deprecations work as expected
• added a 'warning' color for badges for consistency with other statuses
• updated icon demo for new badge 'warning' color
• updated icon-set demo page to use new attribute-based API
• added isStringOrNil to common utils
• added hasPropertyChangedAndNotNil to common utils
• changed behavior in icon element so that setting size to nil or empty string clears out sizing
• preserved behavior in icon element where setting shape to nil or empty string accidentally is ignored
• expanded the icon sizing classname vs. styles helpers to handle new edge cases
• added tests to improve code coverage of the icon classname helpers
• added SizeUpdateStrategies enum and a less awesome pattern-matching approach to changing t-shirt vs. numeric sizing
• accounted for null checking with some typescript gymnastics

Signed-off-by: Scott Mathis <smathis@vmware.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

It updates the API of Clarity Core icons to match the newer APIs of the other components.

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [x] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #4193 

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

### CODE REVIEW

File: src/clr-core/common/utils/identity.ts [hunk 1]
```diff
@@ -25,12 +29,18 @@ export function isObjectAndNotNilOrEmpty(val: any) {
...
+  const tests = [() => isStringOrNil(val), () => val !== oldVal];
   const doTests = allPass(tests);
   return doTests(val, oldVal);
 }
 
+// ignores nil values when checking for changes
+export function hasPropertyChangedAndNotNil(val: string, oldVal: string): boolean {
+  return !isNilOrEmpty(val) && hasPropertyChanged(val, oldVal);
+}
+
```
[the remark this answers is not in the document]
Comment (mathisscott) [reply src/clr-core/common/utils/identity.ts]: Yep. And I will do that, thanks!

File: src/clr-core/common/utils/identity.ts [hunk 2]
```diff
@@ -25,12 +29,18 @@ export function isObjectAndNotNilOrEmpty(val: any) {
   return !isNilOrEmpty(val) && isObject(val);
 }
 
+// marks nil values (undefined, null, empty string) as changed
 export function hasPropertyChanged(val: string, oldVal: string): boolean {
-  const tests = [() => !isNilOrEmpty(val), () => isString(val), () => val !== oldVal];
+  const tests = [() => isStringOrNil(val), () => val !== oldVal];
   const doTests = allPass(tests);
   return doTests(val, oldVal);
```
[the remark this answers is not in the document]
Comment (mathisscott) [reply src/clr-core/common/utils/identity.ts]: Yes, it could be simplified. Will do!
